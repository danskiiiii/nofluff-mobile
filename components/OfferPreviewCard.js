import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import React from 'react';

const OfferPreviewCard = ({ onPress, text1, text2 }) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <View style={styles.imageContainer}>
        <Image
          resizeMode="contain"
          style={styles.logo}
          source={{ uri: 'https://nofluffjobs.com/upload/listing/Egnyte_20141107_131448.jpeg' }}
        />
      </View>
      <Text allowFontScaling minimumFontScale={0.5} numberOfLines={1} style={styles.textTitle}>
        {text1} @ {text2}
      </Text>
      <Text numberOfLines={1} style={styles.textLocation}>
        {text1}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ffffff',
    borderLeftWidth: 6,
    borderColor: '#2980b6',
    marginTop: 6,
    marginLeft: 10,
    marginRight: 10,
  },
  imageContainer: {
    alignItems: 'center',
    justifyContent: 'space-around',
    height: 45,
  },
  logo: {
    top: 5,
    width: 60,
    height: 35,
  },
  textTitle: {
    alignSelf: 'center',
    color: 'black',
    fontSize: 16,
    fontWeight: '400',
    paddingTop: 5,
  },
  textLocation: {
    alignSelf: 'center',
    color: 'grey',
    fontSize: 12,
    fontWeight: '100',
    paddingBottom: 10,
  },
});

export default OfferPreviewCard;
