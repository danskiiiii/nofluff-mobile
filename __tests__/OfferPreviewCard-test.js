import 'react-native';
import 'jsdom-global/register';

import { mount, shallow } from 'enzyme';

import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import OfferPreviewCard from '../components/OfferPreviewCard';
import React from 'react';
import renderer from 'react-test-renderer';

describe('OfferPreviewCard snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the OfferPreviewCard', async () => {
    const tree = renderer.create(<OfferPreviewCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

function getComponent(onPress) {
  return <OfferPreviewCard onPress={onPress} />;
}

describe('OfferPreviewCard component', () => {
  it('renders', () => {
    const wrapper = shallow(getComponent());

    expect(wrapper.exists()).toBe(true);
  });

  it('renders Image', () => {
    const wrapper = mount(getComponent());

    expect(wrapper.find('Image').exists()).toBe(true);
  });

  it('renders two text components', () => {
    const wrapper = shallow(getComponent());

    expect(wrapper.find('Text')).toHaveLength(2);
  });

  it('reacts to onPress event', () => {
    const onPress = jest.fn();
    const wrapper = shallow(getComponent(onPress));

    wrapper.find('TouchableOpacity').simulate('press');
    expect(onPress.mock.calls.length).toBe(1);
  });
});
