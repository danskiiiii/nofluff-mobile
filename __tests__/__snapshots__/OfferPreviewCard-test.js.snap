// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferPreviewCard snapshot renders the OfferPreviewCard 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#2980b6",
      "borderLeftWidth": 6,
      "flex": 1,
      "marginLeft": 6,
      "marginRight": 10,
      "marginTop": 6,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 45,
        "justifyContent": "space-around",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 35,
          "top": 5,
          "width": 60,
        }
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    minimumFontScale={0.5}
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "center",
        "color": "black",
        "fontSize": 16,
        "fontWeight": "400",
        "paddingTop": 5,
      }
    }
  >
     @ 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "center",
        "color": "grey",
        "fontSize": 12,
        "fontWeight": "100",
        "paddingBottom": 10,
      }
    }
  />
</View>
`;
